docs/en/advanced_guides/datasets.md

```diff
@@ -1 +1,143 @@
-# Datasets
+# Dataset
+
+In this document, we will introduce functions of `BaseSegDataset` class in MMSegmentation.
+
+## BaseSegDataset
+
+`BaseSegDataset` is a custom dataset class for semantic segmentation task, which is inherited from `BaseDataset` in MMEngine.
+More details of `BaseDataset` could be found in [MMEngine BaseDataset Documentation](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/advanced_tutorials/basedataset.md).
+
+All datasets are inherited from `BaseSegDataset` in the 1.x version of MMSegmentation. Some frequently used methods are defined in `BaseSegDataset`,
+such as getting meta and raw data information of dataset, updating label mapping and palette information when reducing label `0` (which is usually background) and so on.
+
+`BaseSegDataset` is designed for supervised semantic segmentation task thus both images and annotations are necessary.
+The directory structure of corresponding dataset is as below.
+
+```none
+├── data
+│   ├── my_dataset
+│   │   ├── img_dir
+│   │   │   ├── train
+│   │   │   │   ├── xxx{img_suffix}
+│   │   │   │   ├── yyy{img_suffix}
+│   │   │   ├── val
+│   │   │   │   ├── zzz{img_suffix}
+│   │   ├── ann_dir
+│   │   │   ├── train
+│   │   │   │   ├── xxx{seg_map_suffix}
+│   │   │   │   ├── yyy{seg_map_suffix}
+│   │   │   ├── val
+│   │   │   │   ├── zzz{seg_map_suffix}
+
+```
+
+The image and ground truth annotation pair of `BaseSegDataset` should be of the same except suffix.
+A valid image and ground truth annotation filename pair should be like `xxx{img_suffix}` and `xxx{seg_map_suffix}` (extension is also included
+in the suffix). Two keys `data_list` and `metainfo` are introduced in `BaseSegDataset` to save raw data (such as image path)
+and meta data (such as class names of ground truth) information, respectively.
```

Comment:
it would be better to replace these lines with an example to illustrate the item of `data_list` and metainfo